### PR TITLE
poplar: Enable emmc and recovery build support

### PR DESCRIPTION
**Pull request by vchong**
Signed-off-by: Victor Chong <victor.chong@linaro.org>

### CODE REVIEW

**Comment by vchong** [comment]
Depends on:
https://github.com/ARM-software/arm-trusted-firmware/pull/1221
https://github.com/96boards-poplar/l-loader/pull/1

**Comment by vchong** [comment]
@shawnguo2 Please test along with 2 dependencies above and ack, so that Arm can merge this. Thanks!

**Comment by davidcunado-arm** [comment]
jenkins: test this please

**Comment by shawnguo2** [comment]
I tested this PR on my Poplar board with normal and recovery builds.  It works just fine.  Also thanks Victor for adding comments for FIP_SIZE and FIP_BASE_EMMC.  So for this PR:
Acked-by: Shawn Guo <shawn.guo@linaro.org>
Tested-by: Shawn Guo <shawn.guo@linaro.org>

**Comment by vchong** [comment]
All comments addressed, rebased, squashed and tags added.
Please merge.
Thanks!

**Comment by davidcunado-arm** [comment]
jenkins: test this please

**Comment by davidcunado-arm** [comment]
@vchong 
I ran the checks on the earlier version - there quite a few static analysis issues. I'll re-run and distill out the key issues (our filter is still being tweaked!).

**Comment by vchong** [comment]
@davidcunado-arm Sure. Thanks!

File: plat/hisilicon/poplar/include/hi3798cv200.h [hunk 1]
```diff
@@ -63,11 +63,11 @@
 #define EMMC_CLK_50M			(1 << 8)
 #define EMMC_CLK_25M			(2 << 8)
 
-#define EMMC_DESC_SIZE			(0xF0000)
+#define EMMC_DESC_SIZE			(0x00100000) /* 1MB */
```
Comment (davidcunado-arm) [inline plat/hisilicon/poplar/include/hi3798cv200.h]: Static check - Add U / ULL suffix. Could use the U() ULL() macro from utils_def.h

File: plat/hisilicon/poplar/include/platform_def.h [hunk 2]
```diff
@@ -96,6 +95,11 @@
 #endif /* SPD_none */
 #endif
 
+#define POPLAR_EMMC_DATA_BASE 0x02200000
```
Comment (davidcunado-arm) [inline plat/hisilicon/poplar/include/platform_def.h]: Static check - Add U / ULL suffix. Could use the U() ULL() macro from utils_def.h

**Comment by vchong** [comment]
@davidcunado-arm np : ) patches have been updated.

**Comment by davidcunado-arm** [comment]
jenkins: test this please

**Comment by davidcunado-arm** [comment]
jenkins: test this please

**Comment by davidcunado-arm** [comment]
jenkins: test this please

**Pull request merged**